core/server/src/test/java/alluxio/master/file/FileSystemMasterTest.java

```diff
@@ -706,13 +713,30 @@ public void lostFilesDetectionTest() throws Exception {
 
     // run the detector
     HeartbeatScheduler.schedule(HeartbeatContext.MASTER_LOST_FILES_DETECTION);
-    Assert.assertTrue(HeartbeatScheduler.await(HeartbeatContext.MASTER_LOST_FILES_DETECTION, 5,
-        TimeUnit.SECONDS));
+    Assert.assertTrue(HeartbeatScheduler
+        .await(HeartbeatContext.MASTER_LOST_FILES_DETECTION, 5, TimeUnit.SECONDS));
 
     fileInfo = mFileSystemMaster.getFileInfo(fileId);
     Assert.assertEquals(PersistenceState.LOST.name(), fileInfo.getPersistenceState());
   }
 
+  /**
+   * Tests load metadata logic.
+   * @throws Exception if a {@link FileSystemMaster} operation fails
+   */
+  @Test
+  public void testLoadMetadataTest() throws Exception {
+    FileUtils.createDir(Paths.get(mUnderFS).resolve("a").toString());
+    mFileSystemMaster.loadMetadata(new AlluxioURI("alluxio:/a"),
+        LoadMetadataOptions.defaults().setCreateAncestors(true));
+    mFileSystemMaster.loadMetadata(new AlluxioURI("alluxio:/a"),
+        LoadMetadataOptions.defaults().setCreateAncestors(true));
+
+    FileUtils.createFile(Paths.get(mUnderFS).resolve("a/f").toString());
+    mFileSystemMaster.loadMetadata(new AlluxioURI("alluxio:/a"),
+        LoadMetadataOptions.defaults().setCreateAncestors(true).setLoadDirectChildren(true));
+  }
```

Comment:
I see ... we can't simply do getFileInfo because that also loads the metadata ... it might make sense to add an option for getFileInfo and getFileInfoList to control whether we do any loading at all ... that should be a separate PR though